### ISSUE TITLE
/home/seedsync/.ssh error but I'm not using keys.

### ISSUE THREAD

**Issue by quadcom** [ipsingh06/seedsync#63]
2020/07/07 18:29:14 - WARNING - RemoteScanner.RemoteScanner.Sshcp - Command failed: 'mkdir: cannot create directory ‘/home/seedsync’: Permission denied Could not create directory '/home/seedsync/.ssh'. ssh_exchange_identification: read: Connection reset by peer - '
2020/07/07 18:29:14 - WARNING - RemoteScanner.RemoteScanner - Caught an SshcpError: Unknown error - mkdir: cannot create directory ‘/home/seedsync’: Permission denied Could not create directory '/home/seedsync/.ssh'. ssh_exchange_identification: read: Connection reset by peer

**Comment by quadcom** [comment]
And again

Lost connection to remote server. Retrying automatically.
An error occurred while scanning the remote server: 'Unknown error - mkdir: cannot create directory ‘/home/seedsync’: Permission denied Could not create directory '/home/seedsync/.ssh'. ssh_exchange_identification: read: Connection reset by peer'.

kills my connection and restarts the queue.

**Comment by quadcom** [comment]
I should mention I had just updated to 0.83. I didn't see this in 0.82.

**Comment by quadcom** [comment]
2020/07/07 19:11:59 - INFO - RemoteScanner.RemoteScanner - Skipping remote scanfs installation: already installed
2020/07/07 19:11:59 - DEBUG - RemoteScanner.RemoteScanner.Sshcp - Command: ssh -p 22 -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=error -o PubkeyAuthentication=no xxx@xxx.xxx "'/home1/xxxxx/seedsync/scanfs' '/home1/xxxxx/files/1-SYNC'"
2020/07/07 19:12:00 - WARNING - RemoteScanner.RemoteScanner.Sshcp - Command failed: 'mkdir: cannot create directory ‘/home/seedsync’: Permission denied Could not create directory '/home/seedsync/.ssh'. ssh_exchange_identification: Connection closed by remote host - '
2020/07/07 19:12:00 - WARNING - RemoteScanner.RemoteScanner - Caught an SshcpError: Unknown error - mkdir: cannot create directory ‘/home/seedsync’: Permission denied Could not create directory '/home/seedsync/.ssh'. ssh_exchange_identification: Connection closed by remote host
2020/07/07 19:12:00 - DEBUG - RemoteScanner - Process caught an exception
2020/07/07 19:12:00 - ERROR - seedsync.ControllerJob - Caught exception in job ControllerJob
Traceback (most recent call last):
File "/app/python/common/job.py", line 44, in run
self.execute()
File "/app/python/controller/controller_job.py", line 29, in execute
self.__controller.process()
File "/app/python/controller/controller.py", line 199, in process
self.__propagate_exceptions()
File "/app/python/controller/controller.py", line 526, in __propagate_exceptions
self.__remote_scan_process.propagate_exception()
File "/app/python/common/app_process.py", line 124, in propagate_exception
raise exc.re_raise()
File "/app/python/common/app_process.py", line 30, in re_raise
raise self.ee.with_traceback(self.tb)
File "/app/python/common/app_process.py", line 87, in run
self.run_loop()
File "/app/python/controller/scan/scanner_process.py", line 90, in run_loop
files = self.__scanner.scan()
File "/app/python/controller/scan/remote_scanner.py", line 70, in scan
recoverable=recoverable
controller.scan.scanner_process.ScannerError: An error occurred while scanning the remote server: 'Unknown error - mkdir: cannot create directory ‘/home/seedsync’: Permission denied
Could not create directory '/home/seedsync/.ssh'.
ssh_exchange_identification: Connection closed by remote host'.
2020/07/07 19:12:00 - DEBUG - seedsync.ControllerJob - Calling cleanup for ControllerJob
2020/07/07 19:12:00 - DEBUG - seedsync.Controller - Exiting controller
2020/07/07 19:12:00 - ERROR - seedsync - Caught exception
Traceback (most recent call last):
File "/app/python/seedsync.py", line 162, in run
controller_job.propagate_exception()
File "/app/python/common/job.py", line 77, in propagate_exception
raise exc_info[1].with_traceback(exc_info[2])
File "/app/python/common/job.py", line 44, in run
self.execute()
File "/app/python/controller/controller_job.py", line 29, in execute
self.__controller.process()
File "/app/python/controller/controller.py", line 199, in process
self.__propagate_exceptions()
File "/app/python/controller/controller.py", line 526, in __propagate_exceptions
self.__remote_scan_process.propagate_exception()
File "/app/python/common/app_process.py", line 124, in propagate_exception
raise exc.re_raise()
File "/app/python/common/app_process.py", line 30, in re_raise
raise self.ee.with_traceback(self.tb)
File "/app/python/common/app_process.py", line 87, in run
self.run_loop()
File "/app/python/controller/scan/scanner_process.py", line 90, in run_loop
files = self.__scanner.scan()
File "/app/python/controller/scan/remote_scanner.py", line 70, in scan
recoverable=recoverable
controller.scan.scanner_process.ScannerError: An error occurred while scanning the remote server: 'Unknown error - mkdir: cannot create directory ‘/home/seedsync’: Permission denied
Could not create directory '/home/seedsync/.ssh'.
ssh_exchange_identification: Connection closed by remote host'.
2020/07/07 19:12:01 - DEBUG - ActiveScanner - Process received terminate flag
2020/07/07 19:12:01 - DEBUG - ActiveScanner - Exiting process
2020/07/07 19:12:02 - DEBUG - ExtractProcess - Process received terminate flag
2020/07/07 19:12:03 - DEBUG - seedsync.Controller.MPLogger - Stopped listener thread
2020/07/07 19:12:03 - INFO - seedsync.Controller - Exited controller
2020/07/07 19:12:03 - DEBUG - seedsync.ControllerJob - Finished cleanup for ControllerJob
2020/07/07 19:12:03 - DEBUG - seedsync.ControllerJob - Thread ControllerJob stopped
2020/07/07 19:12:27 - DEBUG - seedsync - Persisting states to file
2020/07/07 19:12:57 - DEBUG - seedsync - Persisting states to file
2020/07/07 19:13:27 - DEBUG - seedsync - Persisting states to file
2020/07/07 19:13:57 - DEBUG - seedsync - Persisting states to file
2020/07/07 19:14:27 - DEBUG - seedsync - Persisting states to file

**Comment by ipsingh06** [comment]
What does your docker run config look like?
The `.ssh` directory is created regardless of whether you use password or keys. This likely has to do with the user and group under which you are running the container.

**Comment by quadcom** [comment]
This error looks as though it's on the seedbox side though. I'm running the exact same docker container with the same user account as before. The only difference is I've changed the server address as I've moved back to a different provider

**Comment by quadcom** [comment]
I've confirmed that this is a remote server response. I've changed to a different SFTP server and the error does not happen. 

The file scanner hits a permissions error on the SFTP side and crashes trying to create a file/folder structure for SSH key exchange when the feature is not being used.

Is there any way to disable the creation of those if they are not needed?

I also just confirmed that the directory structure that is trying to be created, /home/seedsync/.ssh, does not exist on the server that does not throw the error. Is it created and deleted immediately?

**Comment by quadcom** [comment]
sudo docker run --name SeedSync -p 8802:8800 -v /volume2/Sync/1-Seedbox/z-config:/config -v /volume2/Sync/1-Seedbox/1-Incoming:/incoming --user 1039:65539 ipsingh06/seedsync:latest

**Comment by quadcom** [comment]
Changed back to the new host and the error returns.

2020/07/07 20:52:02 - WARNING - RemoteScanner.RemoteScanner.Sshcp - Command failed: 'mkdir: cannot create directory ‘/home/seedsync’: Permission denied Could not create directory '/home/seedsync/.ssh'. ssh_exchange_identification: read: Connection reset by peer - '
2020/07/07 20:52:02 - WARNING - RemoteScanner.RemoteScanner - Caught an SshcpError: Unknown error - mkdir: cannot create directory ‘/home/seedsync’: Permission denied Could not create directory '/home/seedsync/.ssh'. ssh_exchange_identification: read: Connection reset by peer

**Comment by ipsingh06** [comment]
The errors are all occurring on the local side, not the remote server. It's not trying to create any directories on the remote server. The relevant part of the error message is "Connection reset by peer". Have you verified that you can connect to the new provider via ssh using the password?

**Comment by quadcom** [comment]
Yes, I can log in SSH. I download stuff all the time. I also can connect with FZ no problem. Keep in mind, like I wrote it's ONLY with this host. It doesn't happen with other hosts. That's what makes me think this is a host-sided issue not being able to create a folder on the host side, not on my Docker container or Docker container host (NAS) which are both local. Nothing has changed on the local side in any way. The only thing that's changed is the Server Address and directory settings in SS config.

But this pops up now and then and kills the LFTP connections. It then starts a new LFTP connection again most often starting a new seed that was waiting in the queue. So I end up with multiple partially DL'd seeds sitting in the queue.

**Comment by ipsingh06** [comment]
Could you please try manually running the scp command that is failing:
`ssh -p 22 -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=error -o PubkeyAuthentication=no xxx@xxx.xxx "'/home1/xxxxx/seedsync/scanfs' '/home1/xxxxx/files/1-SYNC'"`

You should grab the command from your debug log output and run is exactly.

**Comment by quadcom** [comment]
Emailed....

**Comment by ipsingh06** [comment]
@quadcom Were you able to figure out the reason for this issue?

**Comment by quadcom** [comment]
I moved off of Ultraseedbox.com to Whatbox.ca. Still have a server in the Netherlands. They just have their systems set up right.

I don't have those issues anymore.

One thing I will point out though as I discovered setting up whatbox. The path for the scanner or remote folder can't have a preceding '/' or things will error out. That's not the case for the local directory field though. That does have a preceding /.

![image](https://user-images.githubusercontent.com/7085931/89721996-c1146700-d9b1-11ea-80ae-e34eca5fbca8.png)


Perhaps some kind of error catching for the field entry?
